src/gen-ts.ts

```diff
@@ -369,8 +415,8 @@ function handleProperties(analyzerProperties: Iterable<analyzer.Property>):
  * Convert the given Analyzer methods to their TypeScript declaration
  * equivalent.
  */
-function handleMethods(analyzerMethods: Iterable<analyzer.Method>):
-    ts.Method[] {
+function handleMethods(
+    analyzerMethods: Iterable<analyzer.Method>, isStatic = false): ts.Method[] {
```

Comment:
Rather than a primitive boolean, I would make this an object argument. That makes it extensible for the future. E.g. `{isStatic: true}`. That makes the call-side more descriptive too.

Reply:
I wouldn't worry too much about extensibility for private functions, as refactoring is cheap and easy.

+1 to the legibility argument though.

Reply:
Done, added an `opts` to this method.